File: Week 6 Python/mario.py
from cs50 import get_int


def main():

    # 詢問高度，n < 1 時重複詢問
    while True:
        n = get_int("Height: ")
        if n > 0 and n < 9:
            break

    # 打出金字塔
    for i in range(n):
        print_space(n - i - 1)
        print_brick(i + 1)
        print()


# 打出空格的函數
def print_space(length):
    for i in range(length):
        print(" ", end="")


# 打出磚塊的函數
def print_brick(length):
    for i in range(length):
        print("#", end="")


main()

File: Week 1 C/Mario.c
// Implement a program in C that recreates that pyramid, using hashes (#) for bricks,

#include <cs50.h>
#include <stdio.h>

void print_space(int length);
void print_brick(int length);

int main(void)
{
    // 詢問高度 n，n < 1 時重複詢問
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);

    // 打出金字塔
    for (int i = 0; i < n; i++)
    {
        print_space(n - i + 1);
        print_brick(i + 1);
        printf("\n");
    }
}

// 空格函數 print_space
void print_space(int length)
{
    for (int i = 0; i < length; i++)
    {
        printf(" ");
    }
}

// 磚塊函數 print_brick
void print_brick(int length)
{
    for (int i = 0; i < length; i++)
    {
        printf("#");
    }
}
